lib/puppet/util/windows/registry.rb

```diff
@@ -31,16 +31,24 @@ def open(name, path, mode = KEY_READ | KEY64, &block)
     end
 
     def values(subkey)
+      orig_enc = Encoding.default_internal
+      Encoding.default_internal = Encoding::UTF_8
```

Comment:
Seems dangerous to rely on the code page since we have no control over what it is.

[1] pry(main)> Encoding.locale_charmap
=> "CP437"
[2] pry(main)> .chcp 1252
Active code page: 1252
[3] pry(main)> Encoding.locale_charmap
=> "CP1252"


It seems that the ruby developers should have set the locale to Encoding.default_external, not internal (which is `nil` by default).